dsl/filter.go

```diff
@@ -0,0 +1,25 @@
+package dsl
+
+import (
+	"github.com/quasilyte/go-ruleguard/dsl/types"
+)
+
+// VarFilterContext carries Var and environment information into the filter function.
+// It's an input parameter type for the Var.Filter function callback.
+type VarFilterContext struct {
+	// Type is mapped to Var.Type field.
+	Type types.Type
+}
+
+// SizeOf returns the size of the given type.
```

Comment:
Maybe add a link to https://golang.org/pkg/go/types/#StdSizes to explain the assumptions about the size.

Reply:
It uses context-bound sizes info:
https://github.com/quasilyte/go-ruleguard/blob/9729ea74264f267990f6e6fc0cc80a92f0f6f241/ruleguard/ruleguard.go#L30

So it depends on what integrating code is assigned to that field. It could StdSizes, but I would assume most use cases would have something like this:

https://github.com/golang/tools/blob/d88ec18c69c70d818214bd2c98f3ce431ea4fd62/go/analysis/unitchecker/unitchecker.go#L226

Reply:
Added this line to a comment:
```
// It uses the ruleguard.Context.Sizes to calculate the result.
```